src/transformers/cache_utils.py

```diff
@@ -677,6 +675,8 @@ def update(
             self._seen_tokens += key_states.shape[-2]
 
         if len(self.key_cache) <= layer_idx:
+            raise ValueError("QuantizedCache does not support model usage where layers are skipped. Use DynamicCache.")
+        elif len(self.key_cache) <= layer_idx:
```

Comment:
Quantized cache can support anything that dynamic cache supports, and this check should break any generation with quantized cache because we are no longer able to fill in the cache from zero

If mllama fails with quantized cache, maybe we can do `_supports_quantized_cache=False`  and add a TODO for us? Skipping layers here should be also straightforward I hope

Reply:
Good catch!

- in the commit you reviewed this line, tests should be failing
- pushed a commit that fixed it (`py.test tests/models/llama/ -k test_generate_with_quant_cache -vv` now passes)
- this probably means, as you wrote, that quanto tests are not being run. will investigate